File: app/assets/javascripts/channels/tournaments.js
$(function() {
  if ($('.ws_sender').attr('id') == "smoke-tournaments") {
    App.tournaments = App.cable.subscriptions.create("TournamentsChannel", {
      received: function() {
        function_handler_start("Run Smoke test Tournament")
        function_handler_done("Done Smoke test Tournament")

        function_stop_stop("KILL PROCCESS TEST")
      }
   });

    $('.pid_process').click(function(){
      $('.stop').click()
      App.tournaments.send(JSON.stringify({
        pid_cucumber: "pid_cucumber"
      }))
    });

};

    $('#smoke-tournaments').click(function(){

      $(".step_cucumber").click();
      $(".start").click();
      window.scrollTo(0, 999999);

      App.tournaments.send(JSON.stringify({
        smoke_tournaments: true,  //!!!!!
        sleep_retried: $('#generator_sleep_retried').val(),
        between: $('.count_between').val(),
        count_cashier: $('.count_cashier').val(),
        count_cashier_dep: $('.count_cashier_dep').val(),
        count_payouts: $('.count_payouts').val(),
        count_payouts_sum: $('.count_payouts_sum').val(),
        count_deals: $('.count_deals').val(),
        count_deals_sum: $('.count_deals_sum').val(),
        count_pay_approve: $('.count_pay_approve').val(),
        bonus_dep: $("#generator_scenario_cancel_bonus").prop('checked'),
        turbo: $("#generator_scenario_turbo").prop('checked'),
        binary: $("#generator_scenario_binary").prop('checked'),
        count_features: get_data(),
        thread: $('#generator_thread').val(),
        yopmail : $('#generator_scenario_yopmail').prop('checked'),
        clear_cache: $('#generator_clear_clear_cache').prop('checked'),
        geo: $('#generator_geo').val(),
        proxy_server: $('#generator_proxy_server').val(),
        proxy_http: "--proxy-server=" + $('#generator_proxy_http').val(),
        data_from: '"' + $('#generator_data_tournaments_from').val() + '"',
        data_to: '"' + $('#generator_data_tournaments_to').val() + '"',
        ip_address: App.config.ip,
        param_headless: $('#generator_param_headless').val(),
        count_tournaments: $("#generator_count").val()
      }));

      return false;
  })
});

File: app/channels/ws/crm/tournaments.rb
require "./features/helpers/decor"

module Server
  class Tournaments
    attr_accessor :smoke_tournaments, :count_features,
                  :yopmail, :sleep_retried,

                  :count_cashier, :count_cashier_dep, :bonus_dep,
                  :count_payouts, :count_payouts_sum,
                  :count_deals, :count_deals_sum, :between, :turbo, :binary,
                  :count_pay_approve, :data_from, :data_to,

                  :thread, :geo, :clear_cache, :proxy_http,
                  :proxy_server, :ip_address, :param_headless, :count_tournaments

    def initialize(params)
      @params = params

      params.each { |key, value| public_send("#{key}=", value) }
    end

    def get_link
      JSON.parse(@params.to_json).map do |key, value|
        key == "count_features" ? value.to_s : "#{key}=#{value.to_s}"
      end.join(" ")

    end

    def message
      thread.to_i.times do
        if clear_cache == "true"
          puts "Clear cache".yellow
          system "rake tmp:cache:clear"
          system "rm -rf tmp"
          puts "Cache clear done".yellow
        end
        puts
        puts get_link.yellow
        puts;
        send_broadcast "smoke_binomo", "Run Smoke test Tournament"
        system get_link
        send_broadcast "smoke_binomo", "Done Smoke test Tournament"
      end
    end

  end
end
